core/src/main/scala/com/salesforce/op/stages/impl/tuning/DataCutter.scala

```diff
@@ -110,26 +111,28 @@ class DataCutter(uid: String = UID[DataCutter]) extends Splitter(uid = uid) with
   private[op] def estimate(labelCounts: DataFrame): (Set[Double], Set[Double]) = {
     val minLabelFract = getMinLabelFraction
     val maxLabels = getMaxLabelCategories
+    val maxDroppedLabelsRetained = getMaxDroppedLabelsRetained
 
     val colCount = labelCounts.columns(1)
     val totalValues = labelCounts.agg(sum(colCount)).first().getLong(0).toDouble
-    val labelsKeep = labelCounts
+    val labelsOrderedByCount = labelCounts
       .filter(r => (r.getLong(1) / totalValues) >= minLabelFract)
       .sort(col(colCount).desc)
-      .take(maxLabels)
-      .map(_.getDouble(0))
-
-    val labelSet = labelsKeep.toSet
-    val labelsDropped = labelCounts.filter(r => !labelSet.contains(r.getDouble(0))).collect().map(_.getDouble(0)).toSet
+      .limit(maxLabels + maxDroppedLabelsRetained)
```

Comment:
Is the main improvement here that this sort/limit is "smart" in the sense that it knows not to sort the entire dataframe, but only find the top (maxLabels + maxDroppedLabelsRetained)?